logs/querysample/tracing.go

```diff
@@ -32,6 +34,41 @@ func urlToSample(server *state.Server, grant state.GrantLogs, sample state.Postg
 	)
 }
 
+func startAndEndTime(traceState trace.TraceState, sample state.PostgresQuerySample) (startTime time.Time, endTime time.Time) {
+	if pganalyzeState := traceState.Get("pganalyze"); pganalyzeState != "" {
+		// A pganalyze traceState allows the client to pass the query start time (sent time)
+		// on the client side, in nano second precision, like pganalyze=t:1697666938.6297212
+		// If there are multiple values in a pganalzye traceState, they are separated by semicolon
+		// like pganalyze=t:1697666938.6297212;x=123
+		for _, part := range strings.Split(strings.TrimSpace(pganalyzeState), ";") {
+			if strings.Contains(part, ":") {
+				keyAndValue := strings.SplitN(part, ":", 2)
+				if strings.TrimSpace(keyAndValue[0]) == "t" {
+					if startInSec, err := strconv.ParseFloat(keyAndValue[1], 64); err == nil {
```

Comment:
I really feel like there should be some easy way to do this, I might be simply missing something.
Also, I _could_ try some another way, like splitting `keyAndValue[1]` with `.` then parsing both of them to int. In that way, I don't need to fight with float in test.

Reply:
+1 for parsing this as two separate values. It looks like Go's `time.Time` has an [internal representation](https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/time/time.go;l=129-142) that allows it more nanosecond precision than can be stored in a 64-bit float. I don't think it will _really_ matter in practice, but we might as well get this right. And simplifying the test is a nice bonus.

Since that's going to complicate this branch, I think we should extract that to a separate function like `func parseTraceStateStartTime(value string) (time.Time, error)` (no strong feelings on the name, just throwing something out there).

Reply:
Ah @lfittl sorry looks like you just approved (forgot to push) but I did some refactoring around here and put that functionality to util. Well hope you're okay with that? Or feel free to 👀 again.

Reply:
@keiko713 Refactoring looks good!